lib/router.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+const parser = require('./parse-request');
+const router = module.exports = {};
+const send = require('./sendResponse');
+// Route Registry
+// Store route handlers for each of the verbs ...
+/*
+    i.e.
+    GET: {
+        "/api/note" : (req, res) => {},  ....
+    }
+*/
+router.routeHandlers = {
+  get: {},
+  put: {},
+  post: {},
+  patch: {},
+  delete: {},
+};
+
+Object.keys(router.routeHandlers).forEach((method) => {
+  router.routeHandlers[method] = function(path, callback) {
+    router.routeHandlers[method][path] = callback;
+  };
+});
+
+
+router.route = (req, res) => {
+  //parse the request
+  parser(req).then(request => {
+    if (!request.method) {
```

Comment:
probably an unnecessary error check. If it's coming from the node http module node will already be making this check somewhere.